pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -80,14 +80,16 @@ type Client struct {
 	logger           *log.Scope
 
 	// namespacesFilter is only used to initiate filtered informer.
-	namespacesFilter func(obj interface{}) bool
+	namespacesFilter    func(obj interface{}) bool
+	fieldSelectorsByGVK map[config.GroupVersionKind]string
```

Comment:
suggest use map[config.GroupVersionKind]kubetypes.Filter here, so lately we may need other filter condition like label selector

Reply:
I think the use case is by revision

Reply:
```go
filter := kubetypes.Filter{
    ObjectFilter: func(t any) bool {
        if cl.namespacesFilter != nil && !cl.namespacesFilter(t) {
            return false
        }
        return config.LabelsInRevision(t.(controllers.Object).GetLabels(), cl.revision)
    },
}
```

It appears that currently the namespace filter and revision filter are already supported through separate configurations, but only on the client side.

Reply:
updated

Reply:
good